providers/azure/helpers.go

```diff
@@ -94,3 +83,17 @@ func getContainerClient(conf Config, wrapRoundtripper func(http.RoundTripper) ht
 
 	return containerClient, nil
 }
+
+func getTokenCredential(conf Config) (azcore.TokenCredential, error) {
+	if conf.UserAssignedID == "" {
+		return azidentity.NewDefaultAzureCredential(nil)
+	}
+
+	if conf.ClientSecret != "" && conf.TenantID != "" && conf.ClientID != "" {
+		return azidentity.NewClientSecretCredential(conf.TenantID, conf.ClientID, conf.ClientSecret, &azidentity.ClientSecretCredentialOptions{})
+	}
```

Comment:
Since UserAddignedID & ClientSecret cannot be used at the same time doesn't the current implementation make it so the code in line 93 is never executed because we will always exit on 89, when `UserAssignedID` is not defined?

Reply:
good catch. I'll revert the order and add a test covering it :D